### PR TITLE
SSCS-11486 Clear interloc referral reason

### PR DESCRIPTION
**Pull request by paul-pearson**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/SSCS-11486

### Change description ###

Clear the interloc referral reason when
issuing a new direction or decision

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

**Review by jarekPierchala — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/decisionissued/DecisionIssuedAboutToSubmitHandler.java [hunk 1]
```diff
@@ -54,6 +54,8 @@ public PreSubmitCallbackResponse<SscsCaseData> handle(CallbackType callbackType,
 
         SscsCaseData caseData = callback.getCaseDetails().getCaseData();
 
+        clearInterlocReferralReason(caseData);
```
Comment (nickhill111) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/decisionissued/DecisionIssuedAboutToSubmitHandler.java:57]: I would put a flag around this and check what has been selected from the drop down, from this ticket it seems like we should only clear it if one of the post hearing ones has been selected

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/directionissued/DirectionIssuedAboutToSubmitHandler.java [hunk 2]
```diff
@@ -298,6 +298,8 @@ private PreSubmitCallbackResponse<SscsCaseData> buildResponse(Callback<SscsCaseD
                                                                   DocumentLink url,
                                                                   SscsDocumentTranslationStatus documentTranslationStatus) {
 
+        clearInterlocReferralReason(caseData);
```
Comment (nickhill111) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/directionissued/DirectionIssuedAboutToSubmitHandler.java:301]: I would put a flag around this and check what has been selected from the drop down, from this ticket it seems like we should only clear it if one of the post hearing ones has been selected

**Review by nickhill111 — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/decisionissued/DecisionIssuedAboutToSubmitHandler.java (hunk 1 above)
Comment (paul-pearson) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/decisionissued/DecisionIssuedAboutToSubmitHandler.java]: PH A flag added.  Makeda has confirmed that any pre-existing reason should be cleared, not just PH ones.

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/directionissued/DirectionIssuedAboutToSubmitHandler.java (hunk 2 above)
Comment (paul-pearson) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/directionissued/DirectionIssuedAboutToSubmitHandler.java]: PH A flag added.  Makeda has confirmed that any pre-existing reason should be cleared, not just PH ones.

**Review by nickhill111 — APPROVED** [review]

**Review by nickhill111 — DISMISSED** [review]

**Review by nickhill111 — APPROVED** [review]

**Review by DGordonV1 — APPROVED** [review]

**Pull request merged**